docs/third-party-developers/extensibility/rest-api/available-endpoints-to-extend.md

```diff
@@ -2,74 +2,175 @@
-
-### Key
-
--   `CartSchema::IDENTIFIER`
-
-## `wc/store/cart/items`
-
-The items endpoint, which is also available on `wc/store/cart` inside the `items` key. The data would be available inside each item of the `items` array.
-
-### Passed Parameters
-
--   `data_callback`: `$cart_item`.
--   `schema_callback` none.
-
-### Key
-
--   `CartItemSchema::IDENTIFIER`
-
-## `wc/store/products`
-
-The main products endpoint is extensible via ExtendSchema. The data is available via the `extensions` key for each `product` in the response array.
-
-### Passed Parameters
-
--   `data_callback`: `$product`.
--   `schema_callback` none.
-
-### Key
-
--   `ProductSchema::IDENTIFIER`
+-   [Checkout](#checkout)
+    -   [Use Cases](#use-cases)
+    -   [Example](#example)
+-   [Cart](#cart)
+    -   [Use Cases](#use-cases-1)
+    -   [Example](#example-1)
+-   [Cart Items](#cart-items)
+    -   [Use Cases](#use-cases-2)
+    -   [Example](#example-2)
+-   [Products](#products)
+    -   [Use Cases](#use-cases-3)
+    -   [Example](#example-3)
+
+Some endpoints of the Store API are extensible via a class called `ExtendSchema`. This allows you to customise the data (including the schema) that is returned by the Store API so that it can be consumed by your application or plugin.
+
+For more information about extending the Store API, you may also be interested in:
+
+-   [The full list of available endpoints, including those that cannot currently be extended](../../../internal-developers/rest-api/extend-rest-api.md)
+-   [How to add your data to Store API using `ExtendSchema`](./extend-rest-api-add-data.md)
+-   [How to add a new endpoint](../../../internal-developers/rest-api/extend-rest-api-new-endpoint.md)
+
+Below is a list of available endpoints that you can extend using `ExtendSchema`, as well as some example use-cases.
+
+## Checkout
+
+The `wc/store/checkout` endpoint is extensible via ExtendSchema. Additional data is available via the `extensions` key in the response.
+
+This endpoint can be extended using the `CheckoutSchema::IDENTIFIER` key. For this endpoint, your `data_callback` and `schema_callback` functions are passed no additional parameters.
+
+### Use Cases
+
+This endpoint is useful for adding additional data to the checkout page, such as a custom payment method which requires additional data to be collected from the user or server.
+
+Important: Do **not** reveal any sensitive data in this endpoint, as it is publicly accessible. This includes private keys for payment services.
+
+### Example
+
+```php
+woocommerce_store_api_register_endpoint_data(
+	array(
+		'endpoint' => CheckoutSchema::IDENTIFIER,
+		'namespace' => 'my_plugin_namespace',
+		'data_callback' => function() {
+			return array(
+				'foo' => 'bar',
+			);
+		},
+		'schema_callback' => function() {
+			return array(
+				'properties' => array(
+					'foo' => array(
+						'type' => 'string',
+					),
+				),
+			);
+		}
+		'schema_type' => ARRAY_A,
+	)
+);
```

Comment:
```suggestion
woocommerce_store_api_register_endpoint_data(
	array(
		'endpoint'        => CartItemSchema::IDENTIFIER,
		'namespace'       => 'my_plugin_namespace',
		'data_callback'   => function( $cart_item ) {
			$product = $cart_item['data'];
			return array(
				'my_meta_data' => get_post_meta( $product->get_id(), 'my_meta_data', true ),
			);
		},
		'schema_callback' => function() {
			return array(
				'properties' => array(
					'my_meta_data' => array(
						'type' => 'string',
					),
				),
			);
		},
		'schema_type'     => ARRAY_A,
	)
);
```

Should we consider using PHPCS to enhance readability and set a good example? This feedback is applicable to the other sections as well.